feat: implemented pokemon detail view page

dapeventigan opened the pull request:
- added pokemon detail vm,connector, and page
- displayed pokemon abilities,moves, and stats
- added a dispose when exiting pokemon detail page
- fixed constants

[Working App]
https://user-images.githubusercontent.com/112401492/228272937-30aae302-00ae-4cb9-8c99-5166e20c7596.mp4

[Flutter analyze and Dart Format]
![image](https://user-images.githubusercontent.com/112401492/228273114-39422523-e1a0-4af6-902c-0644160b3aa2.png)

dapeventigan committed:
feat: added pokemon detail view page

- added pokemon detail vm,connector, and page
- displayed pokemon abilities,moves, and stats
- added a dispose when exiting pokemon detail page
- fixed constants

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_details/pokemon_details_connector.dart:
```diff
@@ -0,0 +1,31 @@
+      vm: () => PokemonDetailsVmFactory(),
+      onInit: (store) => store.dispatchAsync(GetPokemonDetailsAction(pokemonName: pokemonName)),
+      onDispose: (store) => store.dispatchSync(ClearPokemonDetailsAction()),
+      builder: (context, vm) => PokemonDetailsPage(
```
kindly rename context to _ since it's unused 🙂

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_details/pokemon_details_page.dart:
```diff
@@ -0,0 +1,134 @@
+  }) : super(key: key);
+
+  final Async<PokemonDetails> pokemonDetails;
+  final String pokeName;
```
pass the whole pokemon object here instead and also place it above `pokemonDetails` property 🙂

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_details/pokemon_details_page.dart:
```diff
@@ -0,0 +1,134 @@
+  final Async<PokemonDetails> pokemonDetails;
+  final String pokeName;
+  final String pokemonImage;
+  @override
```
kindly add a new line above this 🙂

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_details/pokemon_details_page.dart:
```diff
@@ -0,0 +1,134 @@
+          child: Center(
+            child: Column(
+              children: [
+                const SizedBox(height: 30),
```
kindly create a reusable widgets for this kind of scenario name those as:

- VerticalSpace
- HorizontalSpace

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_details/pokemon_details_page.dart:
```diff
@@ -0,0 +1,134 @@
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    children: [
+                      const Icon(MaterialCommunityIcons.pokemon_go, color: appDefaultColor),
+                      const SizedBox(width: 8),
```
kindly make its value double since it's basically asking for a double value 🙂 

same goes to other value assignment you made for this pr 🙂

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_details/pokemon_details_page.dart:
```diff
@@ -0,0 +1,134 @@
+
+  void _showErrorMessageSnackbar(BuildContext context, String message) {
+    ScaffoldMessenger.of(context).showSnackBar(
+      SnackBar(content: Text(message)),
```
no need for this extra trailing comma 🙂

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_details/widget/pokemon_ability.dart:
```diff
@@ -0,0 +1,57 @@
+    Key? key,
+  }) : super(key: key);
+
+  final List<PokemonAbilities>? abilities;
```
haven't I explained before in terms of collection objects, as much as possible never resort to make them as nullable. Instead make them non-nullable then pass an empty list instead if ever 🙂

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_details/widget/pokemon_general.dart:
```diff
@@ -0,0 +1,29 @@
+  Widget build(BuildContext context) {
+    return Chip(
+      label: Text(
+        "$detailName: ${detailValue.toString().split(".").first} $detailType",
```
kindly wrap this in a new variable with better name, make it readily available before the `return` statement 🙂

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_details/widget/pokemon_moves.dart:
```diff
@@ -0,0 +1,60 @@
+    Key? key,
+  }) : super(key: key);
+
+  final List<PokemonMoves>? moves;
```
see related feedback above about this 🙂

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_details/widget/pokemon_stats.dart:
```diff
@@ -0,0 +1,67 @@
+    Key? key,
+  }) : super(key: key);
+
+  final List<PokemonStats>? stats;
```
see related feedback above about this 🙂

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_details/widget/pokemon_stats.dart:
```diff
@@ -0,0 +1,67 @@
+                    final stat = stats?[index].stat;
+                    final baseStat = stats?[index].baseStat;
+                    return Text(
+                      "${stat?.name.capitalize()} : ${baseStat.toString().split('.').first}",
```
kindly wrap this in a new variable with better name, make it readily available before the `return` statement 🙂

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_overview/widget/pokemon_overview_page_card.dart:
```diff
@@ -0,0 +1,46 @@
+        onTap: () => Navigator.push(
+          context,
+          MaterialPageRoute(
+              builder: (context) =>
```
kindly rename context to _ since it's unused 🙂

shawn-ffufm commented on pokedex_async_redux/lib/features/pokemon_overview/widget/pokemon_overview_page_card.dart:
```diff
@@ -0,0 +1,46 @@
+          context,
+          MaterialPageRoute(
+              builder: (context) =>
+                  PokemonDetailsConnector(pokemonName: pokemon.name, pokemonImage: pokemonImageUrlPath)),
```
why not just pass the whole pokemon object instead since you used its entire properties 🙂

shawn-ffufm reviewed (changes requested):
see pr feedback 🙂

dapeventigan committed:
feat: added pokemon detail view page

- made sizebox widgets resuable
- passed pokemon object instead of pokemon.name only
- made collection object non-nullable
- fixed unused context and unnecessary trailing commas
- fixed readability of string values
